README.md

```diff
@@ -8,20 +8,432 @@
+
+Managing state is hard. React allows you to encapsulate two different "kinds" of
+state: "internal" state that can only be mutated and accessed by the component
+that manages it (`state`) and date that gets passed down to the component that
+renders it (`props`). This sounds way more complicated than it actually is, so
+let's take a step back and look at a simple example: A blog. The following
+examples are going to concentrate on a `Blog` component that renders multiple
+`BlogPost`s. Visually, this might look something like this:
+
+![Blog](./diagrams/overview.png)
+
+### Pure Components
+
+React is all about rendering _stuff_: A React component takes some data, builds
+some virtual representation of how our DOM is *supposed* to look like and
+returns it:
+
+```js
+class BlogPost extends React.Component {
+  render () {
+    return (
+      <article>
+        <h1>{this.props.title}</h1>
+        <p>{this.props.body}</p>
+      </article>
+    );
+  }
+}
+```
+
+The above `BlogPost` component renders an article consisting of a `title` and
+`body`. Those are being passed down as `props`. So whenever we want to re-use
+the `BlogPost` component, we can simply give it a `title` and `body`:
+
+```js
+class Blog extends React.Component {
+  render () {
+    return (
+      <div>
+        <BlogPost title={'Hello World!'} body={'Hello, this is my blog.'} />
+        <BlogPost title={'Good bye!'} body={'I\'m busy. I\'m shutting this blog down.'} />
+      </div>
+    );
+  }
+}
+```
+
+This is incredibly powerful, since it allows us to gradually remove complexity
+by re-structuring our application into smaller, easier to test units. `props`
+decouple our components.
+
+### Internal State
+
+As we just saw, props can be seen as arguments for pure components. Pure
+components don't rely on any internal state or side-effects: They simply return
+a virtual representation of how they want to "look" like.
+
+There is only one problem: props are always being "passed down" from the
+component's immediate ancestor. So for example, the `Blog` component passes the
+blog post data to the `BlogPost` component. But where does the blog *data*
+actually come from?
+
+```js
+class Blog extends React.Component {
+  render () {
+    return (
+      <div>
+        <BlogPost title={'Hello World!'} body={'Hello, this is my blog.'} />
+        <BlogPost title={'Good bye!'} body={'I\'m busy. I\'m shutting this blog down.'} />
+      </div>
+    );
+  }
+}
+```
+
+In the above example, we simply hard-coded the blog titles and bodies. Obviously
+this isn't really what we want, otherwise we might as well just render
+everything on the server — not very *reactive* at all.
+
+Instead we typically want to dynamically load the data from some API endpoint,
+for instance `GET /api/posts`. So whenever we render a new blog component, we
+do a XHR call and load the blog posts that we want to render.
+
+**Remember:** `componentDidMount` is a lifecycle method, as such it is **not**
```

Comment:
We can remove this callout, pure components will be talked about in a more advanced lesson